### DISCUSSION TITLE
Update navigation library and refactor the navigation of template to reflect with new style

### DISCUSSION THREAD

**Discussion by kaungkhantsoe** [nimblehq/android-templates#592] [RFCs]
## Issue

**Previously, with the old navigation library version `2.7.*`, we used to navigate using route strings as follows:**
```
object Splash : BaseAppDestination("splash")

composable(AppDestination.Splash) {
            SplashScreen(
                ...
            )
        }
```

## Solution

**With the new navigation library version `2.8.*`, we can navigate using as follows:**
```
@Serializable
data class Profile(val name: String)

val navController = rememberNavController()

NavHost(navController = navController, startDestination = Profile(name="John Smith")) {
    composable<Profile> { backStackEntry ->
        val profile: Profile = backStackEntry.toRoute()
        ProfileScreen(name = profile.name) }
}
```
As you can see, there have been major changes in navigation and passing parameters between destinations.

For more information, please visit the following links 🙇 
- [Navigation with Compose](https://developer.android.com/develop/ui/compose/navigation)
- [Design your navigation graph](https://developer.android.com/guide/navigation/design)
- [Navigation library version 2.8.0 release note](https://developer.android.com/jetpack/androidx/releases/navigation#2.8.0)

## Who Benefits?
Android developers

## What's Next?

**Comment by hoangnguyen92dn** [comment]
There are some aspects we should consider for the migration 💡

Previously, we created the AppDestination to handle these things:
- Navigation argument with NavArgument
- Parcelable Argument -> This can be deprecated from our current approach since the new nav compose already support this ✅ 
- Send result back and forth between screen -> We need to assess this for particular or full migration.

![image](https://github.com/user-attachments/assets/a6fd6810-850b-4673-b9dd-cdd87cb8c496)
